pkg/credentialprovider/gcp/metadata.go

```diff
@@ -151,12 +165,22 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 // Enabled implements a special metadata-based check, which verifies the
 // storage scope is available on the GCE VM.
 func (g *containerRegistryProvider) Enabled() bool {
-	value, err := credentialprovider.ReadUrl(metadataScopes+"?alt=json", g.Client, metadataHeader)
-	if err != nil {
+	if !onGCEVM() {
 		return false
 	}
+	// Given that we are on GCE, we should keep retrying until the metadata server responds.
+	var (
+		value []byte
+		err   error
+	)
+	for {
+		value, err = credentialprovider.ReadUrl(metadataScopes+"?alt=json", g.Client, metadataHeader)
+		if err == nil {
+			break
```

Comment:
Don't think my previous comment was addressed: this shouldn't run in such a hot loop.